python/ray/serve/backend_state.py

```diff
@@ -52,17 +53,28 @@ def __init__(self, controller_name: str, detached: bool,
         self._shutdown_obj_ref = None
         self._state = ReplicaState.SHOULD_START
 
-    def recover_from_checkpoint(self):
+    def __get_state__(self):
+        clean_dict = self.__dict__.copy()
+        del clean_dict["_actor_handle"]
+        del clean_dict["_startup_obj_ref"]
+        del clean_dict["_shutdown_obj_ref"]
+        return clean_dict
+
+    def __set_state__(self, d):
+        self.__dict__ = d
```

Comment:
let's explicitly set the non-checkpointed fields to `None` here for clarity and consistency with the constructor